### ISSUE TITLE
Usability Improvement: Sessions vs. App Pins

### ISSUE THREAD

**Issue by ho1ger** [nextcloud/server#6365]
In NC 12 the views for sessions and App Pins were merged. Now you can revoke App Pins using a (hidden) context menu from the list of current/previous sessions. This has caused some irritations already, c.f. https://github.com/nextcloud/twofactor_totp/issues/186#issuecomment-327146297 and https://github.com/nextcloud/server/issues/6075#issuecomment-327015970. It seems people do not know/understand/find the App Pins in the sessions list. 

Let me explain my thoughts about this (copied from my post from the other threat):

1) Imho a session and an App Pin are entirely different concepts. For me an App Pin is something quite persistent that typically lives as long as you want to use a device. A session, however, is a communication relationship between a server and a device that is created and destroyed many many times.
2) Furthermore, not all sessions can be associated to an App Pin; only those sessions started by an app that doesn't support TOTP.

Hence, I never got the idea to search App Pins in the sessions menu.

If we want to keep the menus merged, I suggest to mitigate the issue by permanently showing something in the sessions list that indicates that the list has more capabilities than being a list of sessions. However, as I argued, I think the new combination is not really logical and it might confuse people.

**Comment by ChristophWurst** [comment]
Thanks for filing this ticket here, @ho1ger!

**Comment by jancborchardt** [comment]
Maybe this will be resolved when we add icons to the list? Sessions will have device-specific icons, and app pins will have a separate one.

But yeah in general the list should be kept together, sorted by most recently used first.

**Comment by AND51** [comment]
I agree to @ho1ger ; I installed new NC 13 Beta yesterday and expected to find the old, seperate App PIN list, to clean up my PINs. Hovering the mouse over the sessions by accident, I revealed the three-dots menu. This was not intuitive at all.

Acutally, I would vote for separating PINs and sessions again. I tell you why. Please, have a look at my sessions list:

![image](https://user-images.githubusercontent.com/26226303/35482439-6909c0c2-0435-11e8-86b4-990adefaa715.png)

As you can see, I have an iPhone syncing CalDAV and CardDAV. You see the same user-agent, but multiple times. In NC 12 (and I believe also in NC 11) this _list grew incredibly long_. Dozens of entrys from my iPhone – the same iPhone. The same thing happens/happened with Vivaldi. Vivaldi creates different sessions. Here it is "this session" and 11, 12, 12 and 14 hours ago.

If you want to withdraw an App PIN, say Vivaldi, which entry do you click? And what if you want to delete an old App PIN when you have 50+ entries? Therefore, there should be a better UI as @jancborchardt mentioned. I think then the idea of merging both lists is fine.

/// Update:

If I'm right then all Apps which use a PIN won't produce multiple entries. I cleared the whole list and gave everything but the browser a PIN. But at least one question stays, "how will the list look when browsers which don't use PINs pollute the list?"

**Comment by ChristophWurst** [comment]
> list grew incredibly long

Fixed by https://github.com/nextcloud/server/pull/6544
